components/gitpod-protocol/src/protocol.ts

```diff
@@ -1400,25 +1400,29 @@ export interface OAuth2Config {
 }
 
 export namespace AuthProviderEntry {
-    export type Type = "GitHub" | "GitLab" | "Bitbucket" | "BitbucketServer" | string;
+    export type Type = "GitHub" | "GitLab" | "Bitbucket" | "BitbucketServer" | "AzureDevOps" | string;
     export type Status = "pending" | "verified";
+    export interface OAuth2CustomConfig {
```

Comment:
Or maybe here would be a better place for those comments.
Although, now that I see the name of the shape, it's 95% self-explanatory :laughing: